### PR TITLE
Fixed sorting function in takeByTopology

### PR DESCRIPTION
**Pull request by flyingcougar**
Sorting function in takeByTopology used wrong values.
Added test cases for those + fix.

### CODE REVIEW

**Comment by ConnorDoyle** [comment]
cc @balajismaniam

**Comment by balajismaniam** [comment]
LGTM.

File: pkg/kubelet/cpumanager/util.go [hunk 1]
```diff
@@ -114,12 +114,14 @@ func takeByTopology(topo *topology.CPUTopology, availableCPUs cpuset.CPUSet, num
 		func(i, j int) bool {
 			// Compute the number of CPUs on the same socket as i and j in
 			// the result.
-			iSocketScore := topo.CPUtopoDetails.CPUsInSocket(i).Intersection(result).Size()
-			jSocketScore := topo.CPUtopoDetails.CPUsInSocket(j).Intersection(result).Size()
+			iSocket := topo.CPUtopoDetails[cpuIDs[i]].SocketID
+			jSocket := topo.CPUtopoDetails[cpuIDs[j]].SocketID
+			iSocketScore := topo.CPUtopoDetails.CPUsInSocket(iSocket).Intersection(result).Size()
+			jSocketScore := topo.CPUtopoDetails.CPUsInSocket(jSocket).Intersection(result).Size()
```
Comment (ConnorDoyle) [inline pkg/kubelet/cpumanager/util.go:120]: :+1:

File: pkg/kubelet/cpumanager/util.go [hunk 2]
```diff
@@ -114,12 +114,14 @@ func takeByTopology(topo *topology.CPUTopology, availableCPUs cpuset.CPUSet, num
...
 
 			// Compute the number of available CPUs on the same core as i and j.
-			iCoreScore := topoDetails.CPUsInCore(topoDetails[i].CoreID).Size()
-			jCoreScore := topoDetails.CPUsInCore(topoDetails[j].CoreID).Size()
+			iCoreScore := topoDetails.CPUsInCore(topoDetails[cpuIDs[i]].CoreID).Size()
+			jCoreScore := topoDetails.CPUsInCore(topoDetails[cpuIDs[j]].CoreID).Size()
```
Comment (ConnorDoyle) [inline pkg/kubelet/cpumanager/util.go:124]: :+1:

File: pkg/kubelet/cpumanager/util_test.go [hunk 3]
```diff
@@ -75,6 +83,14 @@ func TestTakeByTopology(t *testing.T) {
 			cpuset.NewCPUSet(2, 6),
 		},
 		{
+			"take three cpus from dual socket with HT - core form Socket 1",
```
Comment (ConnorDoyle) [inline pkg/kubelet/cpumanager/util_test.go]: s/form/from

**Review by ConnorDoyle — CHANGES_REQUESTED** [review]
Just one typo, otherwise LGTM

**Comment by flyingcougar** [comment]
done

**Comment by ConnorDoyle** [comment]
Thanks! Squash-merged

**Comment by ConnorDoyle** [comment]
Hey sorry, I had to revert this. It broke the unit tests for the static policy.

```
--- FAIL: TestStaticPolicyRegister (0.00s)
        policy_static_test.go:239: StaticPolicy Register() error (GuPodSingleCore, SingleSocketHT, ExpectAllocOneCPU). expected cpuset 1 but got 4
```

**Comment by ConnorDoyle** [comment]
Think you mentioned this issue offline @flyingcougar -- the expected value for this allocation should have been 4, not 1. The reason is that since cpu 0 is unavailable takeByTopology should select cpu 0's sibling which has ID 4.

**Comment by ConnorDoyle** [comment]
Updated the expected value and re-pushed.

**Pull request merged**